### PR TITLE
Added attribute n_initial_contacts

### LINKED ISSUE #2604: Document Contacts attributes and add n_initial_contacts
## Is your feature request related to a problem? ##
``Contacts`` gives the fraction of initial contacts in ``timeseries``, but I think most people would also like to know how many initial contacts there actually are. 
 
## Describe the solution you'd like ##
You can calculate this with ``Contacts.initial_contacts[0].sum()``, but it would be nice to just have it there as an attribute e.g. `n_initial_contacts`. Especially since the documentation doesn't list ``initial_contacts`` or ``r0`` as attributes, so many users will not know it's there. So, desired solution:

1. Have `n_initial_contacts`
2. Document the (important) attributes of the class

## Additional context ##

[Example notebook of using the class, and calculating initial contacts, in this user guide notebook.](https://www.mdanalysis.org/UserGuide/examples/analysis/distances_and_contacts/contacts_native_fraction.html#Calculating-fraction-of-native-contacts,-with-a-distance-lower-than-or-equal-to-the-reference-structure)

### PR DESCRIPTION
**Pull request by Oscuro-Phoenix**
Fixes #2604

Changes made in this Pull Request:
 - 
Added attribute for total contacts initially in class Contacts

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by Oscuro-Phoenix** [comment]
@IAlibay  Is this fine ?

File: package/MDAnalysis/analysis/contacts.py [hunk 1]
```diff
@@ -438,6 +438,8 @@ def __init__(self, u, select, refgroup, method="hard_cut", radius=4.5,
                 self.r0.append(distance_array(refA.positions, refB.positions))
                 self.initial_contacts.append(contact_matrix(self.r0[-1],
                                                             radius))
+        # Number of initial contacts present
+        self.n_initial_contacts=self.initial_contacts[0].sum()
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/contacts.py]: As mentioned by @lilyminium in #2604 you'll want to document this in the docstring as an attribute that users should be aware of. You'll also want to document things like the `r0` attributes too.

Also please do keep to PEP8 convention, you should have spaces around the `=` here.

Comment (Oscuro-Phoenix) [reply package/MDAnalysis/analysis/contacts.py]: Thanks a lot for reverting @IAlibay, I added the changes suggested by you. Please let me know if its okay. You're right about testing.

File: package/MDAnalysis/analysis/contacts.py [hunk 2]
```diff
@@ -368,11 +368,18 @@ class Contacts(AnalysisBase):
     Attributes
     ----------
     timeseries : list
-        list containing *Q* for all refgroup pairs and analyzed frames
-
+        list containing *Q* for all refgroup pairs and analyzed frames 
+    r0 : list
+        list containing the initial distance array in the reference structure
+    initial_contacts : list
+        list containing the contact_matrix based on the radius cutoff
```
Comment (IAlibay) [inline package/MDAnalysis/analysis/contacts.py]: I'm a bit rusty on the `contacts` code, but I think here you should be saying here that it's the contact matrix based on the reference frame.

File: package/MDAnalysis/analysis/contacts.py [hunk 3]
```diff
@@ -368,11 +368,18 @@ class Contacts(AnalysisBase):
...
     ----------
     timeseries : list
-        list containing *Q* for all refgroup pairs and analyzed frames
-
+        list containing *Q* for all refgroup pairs and analyzed frames 
+    r0 : list
+        list containing the initial distance array in the reference structure
+    initial_contacts : list
+        list containing the contact_matrix based on the radius cutoff 
+    n_initial_contacts : Stores the total number of initial contacts
```
[the remark this answers is not in the document]
Comment (Oscuro-Phoenix) [reply package/MDAnalysis/analysis/contacts.py]: I see, its 'int'. Please check if its resolved.

File: package/MDAnalysis/analysis/contacts.py [hunk 4]
```diff
@@ -368,11 +368,18 @@ class Contacts(AnalysisBase):
...
+        sum of the initial contacts
     .. versionchanged:: 1.0.0
        ``save()`` method has been removed. Use ``np.savetxt()`` on
        :attr:`Contacts.timeseries` instead.
+    .. versionchanged:: 0.21.0
```
[the remark this answers is not in the document]
Comment (Oscuro-Phoenix) [reply package/MDAnalysis/analysis/contacts.py]: I added the stuff under the `versionchanged:: 0.21.0` in `versionchanged:: 1.0.0`.

**Comment by IAlibay** [comment]
@Oscuro-Phoenix it's a lot less work for us to review everything at the same time, because if parts of the code don't work, we could spend extra rounds of review discussing things like docstring only to realise later that it'll need changing (I'm not saying that's the case here, but there are many scenarios where it could happen). 

With this in mind, could you please add the tests before I re-review the PR?

In another note, I just noticed that `rmsfit_adk_dims.dcd` somehow made it into this PR (looks like dc5f855). This shouldn't be there. Since this is related to #2639, any chance you might know how this file is getting generated?

**Comment by Oscuro-Phoenix** [comment]
Thanks a lot, will keep that in mind! Will add the tests before the re-review. I'm sorry!

Well, ya `rmsfit_adk_dims.dcd` may have made its way because I ran a pytest (test_align.py) earlier. I'll delete it.

**Comment by IAlibay** [comment]
@Oscuro-Phoenix is this still active?

**Comment by Oscuro-Phoenix** [comment]
Seems like the related issue hasn't closed yet. Should I continue to fix this?

**Comment by IAlibay** [comment]
> Seems like the related issue hasn't closed yet. Should I continue to fix this?

@Oscuro-Phoenix My message was only to check if you were still interested in completing this as there hadn't been any changes in a long time. Please do continue.

**Comment by lilyminium** [comment]
Hi @Oscuro-Phoenix, are you still keen to work on this?

**Comment by lilyminium** [comment]
Closing due to no reply from @Oscuro-Phoenix for a week. Please let me know if this was in error and you'd still like to work on this.

**Pull request closed without merging**